extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/MSQErrorReport.java

```diff
@@ -195,10 +213,21 @@ public static MSQFault getFaultFromException(@Nullable final Throwable e)
         return new RowTooLargeFault(((FrameRowTooLargeException) cause).getMaxFrameSize());
       } else if (cause instanceof InvalidNullByteException) {
         InvalidNullByteException invalidNullByteException = (InvalidNullByteException) cause;
+        String columnName = invalidNullByteException.getColumn();
+        if (columnMappings != null) {
+          IntList outputColumnsForQueryColumn = columnMappings.getOutputColumnsForQueryColumn(columnName);
+
+          // outputColumnsForQueryColumn.size should always be 1 due to hasUniqueOutputColumnNames check that is done
+          if (outputColumnsForQueryColumn.size() >= 1) {
```

Comment:
nit: Should we handle the case where it is more than 1 separately for neatness? That said, since this is code to handle a different type of exception, I am okay with the current code.

Reply:
I think it should be fine since even if there are multiple columns for a single query column, to surface the exception, we can only display one name, and displaying the first one seems okay.

If the code which throws `InvalidNullByteException` instead showed all the columns with values \0 in them (which I think won't be possible), we can modify it, however as the code currently is, we throw on encountering the first value with \0.